### ISSUE TITLE
Flutter iOS application won't run on simulator

### ISSUE THREAD

**Issue by pranavchunduru3** [flutter/flutter#71240]
Overview: I am trying to run my iOS app(flutter run) and it is getting a many errors that I can't find a solution to. 
flutter create bug:

```
Recreating project bug...                   
Wrote 3 files.

All done!
[✓] Flutter: is fully installed. (Channel stable, 1.22.1, on macOS 11.0.1 20B29, locale en-US)
[!] Android toolchain - develop for Android devices: is partially installed; more components are available. (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS: is fully installed. (Xcode 12.2)
[✓] Android Studio: is fully installed. (version 3.6)
[✓] VS Code: is fully installed. (version 1.51.1)
[✓] Connected device: is fully installed. (1 available)

Run "flutter doctor" for information about installing additional components.

In order to run your application, type:

  $ cd bug
  $ flutter run

Your application code is in bug/lib/main.dart.
```

**Expected results:** 
I wanted to see a running app on the simulator
**Actual results:** 
I get this error: 
```
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/pranavchunduru/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.16.1/darwin/Classes/AudioplayersPlugin.m:89:37: warning: incompatible pointer
    types sending 'FlutterEngine *' to parameter of type 'NSObject<FlutterBinaryMessenger> * _Nonnull' [-Wincompatible-pointer-types]
                        binaryMessenger:_headlessEngine];
                                        ^~~~~~~~~~~~~~~
    In module 'Flutter' imported from /Users/pranavchunduru/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.16.1/darwin/Classes/AudioplayersPlugin.h:2:
    /Users/pranavchunduru/Desktop/cyrano/frontend/ios/Flutter/Flutter.framework/Headers/FlutterChannels.h:178:74: note: passing argument to parameter
    'messenger' here
                          binaryMessenger:(NSObject<FlutterBinaryMessenger>*)messenger;
                                                                             ^
    /Users/pranavchunduru/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.16.1/darwin/Classes/AudioplayersPlugin.m:480:74: warning: 'initWithImage:' is
    deprecated: first deprecated in iOS 10.0 - Use -initWithBoundsSize:requestHandler: [-Wdeprecated-declarations]
                  MPMediaItemArtwork *albumArt = [[MPMediaItemArtwork alloc] initWithImage: artworkImage];
                                                                             ^
    In module 'MediaPlayer' imported from
    /Users/pranavchunduru/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.16.1/darwin/Classes/AudioplayersPlugin.m:7:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.2.sdk/System/Library/Frameworks/MediaPlay
    er.framework/Headers/MPMediaItem.h:240:1: note: 'initWithImage:' has been explicitly marked deprecated here
    - (instancetype)initWithImage:(UIImage *)image MP_DEPRECATED("Use -initWithBoundsSize:requestHandler:", ios(5.0, 10.0));
    ^
    2 warnings generated.
    ld: framework not found Flutter
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
```

<details>
  <summary>Logs</summary>

```
```

```
Analyzing frontend...                                                   

   info • 'autovalidate' is deprecated and shouldn't be used. Use autoValidateMode parameter which provide more specific behaviour related to auto
          validation. This feature was deprecated after v1.19.0. • lib/screens/authenticate/forgot_password.dart:101:7 • deprecated_member_use
   info • 'autovalidate' is deprecated and shouldn't be used. Use autoValidateMode parameter which provide more specific behaviour related to auto
          validation. This feature was deprecated after v1.19.0. • lib/screens/authenticate/sign_in.dart:158:7 • deprecated_member_use
   info • 'autovalidate' is deprecated and shouldn't be used. Use autoValidateMode parameter which provide more specific behaviour related to auto
          validation. This feature was deprecated after v1.19.0. • lib/screens/authenticate/sign_in.dart:170:7 • deprecated_member_use
   info • 'autovalidate' is deprecated and shouldn't be used. Use autoValidateMode parameter which provide more specific behaviour related to auto
          validation. This feature was deprecated after v1.19.0. • lib/screens/authenticate/sign_up.dart:153:7 • deprecated_member_use
   info • 'autovalidate' is deprecated and shouldn't be used. Use autoValidateMode parameter which provide more specific behaviour related to auto
          validation. This feature was deprecated after v1.19.0. • lib/screens/authenticate/sign_up.dart:163:7 • deprecated_member_use
   info • 'autovalidate' is deprecated and shouldn't be used. Use autoValidateMode parameter which provide more specific behaviour related to auto
          validation. This feature was deprecated after v1.19.0. • lib/screens/authenticate/sign_up.dart:175:7 • deprecated_member_use
   info • The method doesn't override an inherited method • lib/services/auth.dart:88:16 • override_on_non_overriding_member

7 issues found. (ran in 5.8s)
```
```
```

```
Pranavs-MacBook-Pro:frontend pranavchunduru$ flutter doctor -v
[✓] Flutter (Channel stable, 1.22.1, on macOS 11.0.1 20B29, locale en-US)
    • Flutter version 1.22.1 at /Users/pranavchunduru/flutter
    • Framework revision f30b7f4db9 (7 weeks ago), 2020-10-08 10:06:30 -0700
    • Engine revision 75bef9f6c8
    • Dart version 2.10.1

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/pranavchunduru/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.10.0

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

 
[✓] Connected device (1 available)            
    • iPhone 12 Pro Max (mobile) • AA869F57-540B-478F-9E52-D565B309A30D • ios • com.apple.CoreSimulator.SimRuntime.iOS-14-2 (simulator)

! Doctor found issues in 1 category.
```

```
```

</details>

**Comment by Abhishek01039** [comment]
Hi @pranavchunduru3 
seems like you are using a third-party plugin.
Please open the issue to appropriate repos.
Thanks.

**Comment by darshankawar** [comment]
@pranavchunduru3 
What command are you using to build app on simulator ?
Does it run through Xcode ?
Check [this](https://github.com/flutter/flutter/issues/43591#issuecomment-547219949) issue comment and see if it works for you.
Thanks.

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
